credentials/java/com/google/auth/Credentials.java

```diff
@@ -70,6 +72,27 @@ public String getUniverseDomain() throws IOException {
     return GOOGLE_DEFAULT_UNIVERSE;
   }
 
+  /**
+   * Gets the credential type used for internal metrics header.
+   *
+   * @return a enum value for credential type
+   */
+  public CredentialTypeForMetrics getMetricsCredentialType() {
+    return this.credentialTypeForMetrics;
+  }
+
+  /**
+   * Sets the credential type for metrics.
+   *
+   * <p>The default is {@code CredentialTypeForMetrics.DO_NOT_SEND}. For a credential that is
+   * established to track for metrics, this default should be overridden.
+   *
+   * @param credentialTypeForMetrics The credential type to be used for metrics.
+   */
+  public void setMetricsCredentialType(CredentialTypeForMetrics credentialTypeForMetrics) {
+    this.credentialTypeForMetrics = credentialTypeForMetrics;
+  }
+
```

Comment:
On second thought, I think it may be possible to remove `setMetricsCredentialType()`? I'd rather not add a public setter and I think we can do it this way:

Credentials' `getMetricsCredentialType()` returns `DO_NOT_SEND` by default:
```
public CredentialTypeForMetrics getMetricsCredentialType() {
    return CredentialTypeForMetrics.DO_NOT_SEND;
}
```

Each subclass would override this value: 
i.e. ComputeEngineCredentials' `getMetricsCredentialType()`:
```
public CredentialTypeForMetrics getMetricsCredentialType() {
    return CredentialTypeForMetrics.VM_CREDENTIALS;
}
```

When we add a new Metrics for Credentials, we'll just need to override this value. Do you think there are any issues with this?

Reply:
Thanks for pointing this out. In fact, I switched away from this approach assuming issues for SA with 2 flow types. But I just took another look and this should work fine.